### PR TITLE
Feature/edit details

### PR DESCRIPTION
**Pull request by YetAnotherMatt**
Refactor the building of summary list rows by offloading the logic to the step definitions. This allows paths (for editing purposes) to easily be associated with each step summary. It also automatically reorders the summary if the apply journey flow changes.
Each page now defines the summary data for the check details page, including how many rows should be displayed. This also means that no changes are required to the check details page when a new page is added to the flow.

### CODE REVIEW

File: src/web/routes/application/are-you-pregnant/are-you-pregnant.js [hunk 1]
```diff
@@ -7,6 +8,25 @@ const exampleDate = (fromDate = new Date()) => {
...
 
+const contentSummary = (req) => {
+  const pregnantSummary = {
+    key: req.t('areYouPregnant.summaryKey'),
+    value: req.t(req.session.claim.areYouPregnant)
+  }
+  const dueDateSummary = {
+    key: req.t('areYouPregnant.expectedDeliveryDateSummaryKey'),
+    value: [
+      req.session.claim['expectedDeliveryDate-day'],
```
Comment (mikeSmith08) [inline src/web/routes/application/are-you-pregnant/are-you-pregnant.js]: You'll need to change this to use the new date formatting.

File: src/web/routes/application/are-you-pregnant/are-you-pregnant.test.js [hunk 2]
```diff
@@ -16,3 +16,53 @@ test('exampleDate() should correctly roll into the next year', (t) => {
...
+  }
+
+  const result = contentSummary(req)
+
+  const expected = [{
+    key: 'areYouPregnant.summaryKey',
+    value: 'yes'
+  }, {
+    key: 'areYouPregnant.expectedDeliveryDateSummaryKey',
+    value: '01 03 2019'
```
Comment (mikeSmith08) [inline src/web/routes/application/are-you-pregnant/are-you-pregnant.test.js]: You'll need to correct this value with the new date formatting.

File: src/web/routes/application/card-address/card-address.js [hunk 3]
```diff
@@ -13,15 +15,30 @@ const pageContent = ({ translate }) => ({
   formDescription: translate('cardAddress.formDescription')
 })
 
+const newLineChar = '\n'
+const addressToString = compose(join(newLineChar), filter(notIsNilOrEmpty))
```
Comment (mikeSmith08) [inline src/web/routes/application/card-address/card-address.js]: This should be renamed, it doesn't really have anything to do with an address.

File: src/web/routes/application/card-address/card-address.test.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+const test = require('tape')
+const { assocPath } = require('ramda')
+const { contentSummary } = require('./card-address')
+
+const req = {
+  t: string => string,
+  session: {
+    claim: {
+      addressLine1: 'Flat b',
```
Comment (mikeSmith08) [inline src/web/routes/application/card-address/card-address.test.js]: Minor - we have these address details stored and reused in `constants.js` already but this is in the acceptance steps package, can/should we reuse this here? I'm not sure what you'd normally do here...

File: src/web/routes/application/are-you-pregnant/are-you-pregnant.js (hunk 1 above)
Comment (YetAnotherMatt) [reply src/web/routes/application/are-you-pregnant/are-you-pregnant.js]: done

**Pull request merged**